api/envoy/config/cluster/redis/redis_cluster.proto

```diff
@@ -0,0 +1,22 @@
+syntax = "proto3";
+
+package envoy.config.cluster.redis;
+
+option java_outer_classname = "RedisClusterProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.config.cluster.redis";
+option go_package = "v2";
+
+import "google/protobuf/duration.proto";
+
+import "validate/validate.proto";
+import "gogoproto/gogo.proto";
+
+message RedisClusterConfig {
```

Comment:
Please add full docs for what this is, docs for each member, and also make sure the docs are fully rendered. It's a little strange how this works but for an example of how this is wired up fully take a look at the history of this file for example PRs: https://github.com/envoyproxy/envoy/commits/master/docs/build.sh e.g. https://github.com/envoyproxy/envoy/commit/9a06dc0777d2809195cb1fc414b05ae7c0660193#diff-3d9d228ad4585ae89df8a6dc3b9081da. You can build the docs and look at them locally via `docs/build.sh`.

Reply:
Added docs and test-generated them for a few cycles per your advice. I found the RBAC config to be the most helpful to follow, for future reference.

In the architecture section I mention topology only, as that is the limit of this PR, and mention the support is experimental. I do not add any caveats about MOVE/ASK per @msukalski or anything else, but do say this is experimental work.